src/devices/sound/gb.cpp

```diff
@@ -351,27 +351,26 @@ void gameboy_sound_device::update_square_channel(struct SOUND &snd, uint64_t cyc
 			}
 		}
 
-		while (cycles > 0)
+		if (cycles & 3)
```

Comment:
Why did you change the condition from `cycles < 4` to `cycles & 3` for this?  Is it absolutely guaranteed that cycles can never be a higher value, e.g. 6 or 7?